ios/tos/tos/ContentView.swift

```diff
@@ -3,19 +3,39 @@ import SwiftUI
 struct ContentView: View {
     @ObservedObject var menuViewModel = MenuViewModel()
     @ObservedObject var orderViewModel = OrderViewModel()
+    @State var showSettings = false
+    @State var itemSelected: Tospb_Item?
 
     var body: some View {
-        GeometryReader { geo in
-            HStack {
-                OrderView(viewModel: orderViewModel)
-                Divider()
-                MenuView(
-                    menuViewModel: menuViewModel,
-                    orderViewModel: orderViewModel
-                )
-                .frame(minWidth: geo.size.width - (geo.size.width / 3))
+        NavigationView {
+            ZStack {
+                GeometryReader { geo in
+                    HStack {
+                        OrderView(viewModel: orderViewModel, itemSelected: self.$itemSelected)
+                        Divider()
+                        MenuView(
+                            menuViewModel: menuViewModel,
+                            orderViewModel: orderViewModel
+                        )
+                        .frame(minWidth: geo.size.width - (geo.size.width / 3))
+                    }
+                }
+                .navigationBarTitle("Menu", displayMode: .inline)
+                .navigationBarItems(trailing: NavigationLink(destination: SettingsView(), isActive: $showSettings) {
+                    Button(action: {
+                        showSettings = true
+                    }) {
+                        Text("Settings")
+                    }
+                })
+                
+                if itemSelected != nil {
+                    PopupMenuView(viewModel: self.orderViewModel, item: self.$itemSelected, editing: Binding.constant(true))
+                        .padding()
+                }
```

Comment:
Lets hold off on this side of editing as well, I like the idea just want to think of a clean way to do it